### PR TITLE
azd template completion checklist

### PR DESCRIPTION
**Pull request by gkulin**
Adds the checklist for an `azd` template to ensure it meets all the necessary requirements including:
- repository management 
- source code structure
- functional requirements
- security requirements

### CODE REVIEW

File: checklist.md
Comment (charris-msft) [inline checklist.md]: I would format this item the same as the others with "**Issues template** - " at the beginning.

File: checklist.md
Comment (charris-msft) [inline checklist.md]: since there are two sentences, the 2nd one should end with a period even though the rest of the text bullets do not.

File: checklist.md
Comment (charris-msft) [inline checklist.md]: need an extra space between the square brackets to make it a checkbox.

File: checklist.md
Comment (charris-msft) [inline checklist.md]: I would remove this "Testing" requirement in favor of the one under the Functional requirements section.

File: checklist.md
Comment (charris-msft) [inline checklist.md:18]: This section should also include the following requirements:
[ ] **Core Modules** - `main.bicep` should reference modules from `core`, copied from [azure-dev](https://github.com/Azure/azure-dev/tree/main/templates/common/infra/bicep).

`cp -r ../azure-dev/templates/common/infra/bicep/core/* infra/core/.`

File: checklist.md
Comment (charris-msft) [inline checklist.md]: I would bold "Microsoft Managed Service Identity" or add **Managed Identity** to the beginning to make sure that it stands out since it's an important requirement.

File: checklist.md
Comment (charris-msft) [inline checklist.md:45]: We also need the following in this section:

- [ ] **security-devops-action** - The application must run microsoft/security-devops-action. [Example](https://github.com/Azure-Samples/azure-search-openai-demo/blob/ab1e4806a176b084b1980a9ee7e1c55bdac1a6d5/.github/workflows/azure-dev-validation.yaml#L25)
  * If unfamiliar with setting up GH Actions, then you can follow the instructions here: [defender-for-cloud/quickstart-onboard-github](
https://learn.microsoft.com/en-us/azure/defender-for-cloud/quickstart-onboard-github)

File: checklist.md
Comment (charris-msft) [inline checklist.md:5]: Is the idea that additional README requirements like the following would go in the README.md, rather than being called out in this checklist?

- [ ] **Prerequisites** - a Prerequisites section should be included. [Example](https://github.com/Azure-Samples/todo-python-mongo-aca#prerequisites)

- [ ] **Architecture** - an architecture diagram and description must be included. [Example](https://github.com/Azure-Samples/todo-python-mongo-aca#application-architecture)

- [ ] **'Open In __' buttons** - must use same "Open in " buttons as the TODO samples. [Example](https://github.com/Azure-Samples/todo-python-mongo-aca/blob/05a7aa59c0253628e293ca0fcc98f35a942df1cc/README.md?plain=1#L24)

- [ ] **Cost Estimation** - a Cost Estimation section should be included. [Example](https://github.com/Azure-Samples/azure-search-openai-demo#cost-estimation)

**Review by charris-msft — CHANGES_REQUESTED** [review]

File: checklist.md
Comment (yungezz) [reply checklist.md]: Official name is `Managed Identity`

File: checklist.md
Comment (kristenwomack) [inline checklist.md]: Somewhere in this line or the template short description, we should use the full name "Azure Developer CLI" to make the connection with `azd` in case this isn't clear and given the research we've done.

File: checklist.md
Comment (kristenwomack) [reply checklist.md:5]: Yes, we can say that explicitly @charris-msft.

File: checklist.md
Comment (kristenwomack) [inline checklist.md:37]: And should there be a plan for periodic testing or adding e2e tests?

**Review by kristenwomack — COMMENTED** [review]
Great progress, @gkulin -- I can do a second review after these changes are implemented.

File: checklist.md
Comment (gkulin) [reply checklist.md:5]: Yeah, since these things are all already included in the README template I figured we didn't need to list them here. But happy to add them back. The only reason I didn't list these individually here is because everything in the template readme is required (unless marked as option), so if I list these 4 sections then we should probably list all 11 which just felt like a lot. Would love to know what you both think though!

File: checklist.md
Comment (gkulin) [reply checklist.md:37]: Definitely! I just think we have to come up with that plan first

File: checklist.md
Comment (charris-msft) [inline checklist.md:53]: Do we have a link/links we can point them to?

**Review by charris-msft — APPROVED** [review]
Approved, but would be nice to have appropriate links for styleguides.

File: checklist.md
Comment (gkulin) [reply checklist.md:53]: we can point to the styleguides from Natalia https://github.com/Azure-Samples/azd-template-artifacts/pull/3/files once her pr is merged!

File: checklist.md
Comment (anfibiacreativa) [reply checklist.md:5]: We can list them, explicitly, yes. Are you ok with us adding a detailed list in a next iteration?

File: checklist.md
Comment (anfibiacreativa) [inline checklist.md:16]: Isn't this for Microsoft authored templates only tho?

**Review by yungezz — APPROVED** [review]

File: checklist.md
Comment (anfibiacreativa) [reply checklist.md:37]: This may in scope of the pipeline automation

**Review by anfibiacreativa — APPROVED** [review]

File: checklist.md
Comment (kristenwomack) [reply checklist.md:16]: Sometimes external authors ask us about metrics.

**Pull request merged**
